tests/topotests/lib/common_config.py

```diff
@@ -1258,6 +1261,26 @@ def verify_rib(tgen, addr_type, dut, input_dict, next_hop=None, protocol=None):
     return True
 
 
+def verify_rib(tgen, addr_type, dut, input_dict, next_hop=None, protocol=None, expected=True):
+    """
+    Wrapper function for `_verify_rib` that tries multiple time to get results.
+
+    When the expected result is `False` we actually should expect for an string instead.
+    """
+
+    # Use currying to hide the parameters and create a test function.
+    test_func = partial(_verify_rib, tgen, addr_type, dut, input_dict, next_hop, protocol)
+
```

Comment:
[update_wrapper](https://docs.python.org/2/library/functools.html#functools.update_wrapper) here is probably useful